activemq-broker/src/main/java/org/apache/activemq/broker/region/Topic.java

```diff
@@ -826,14 +830,61 @@ protected void dispatch(final ConnectionContext context, Message message) throws
     }
 
     private final AtomicBoolean expiryTaskInProgress = new AtomicBoolean(false);
-    private final Runnable expireMessagesWork = new Runnable() {
-        @Override
-        public void run() {
-            List<Message> browsedMessages = new InsertionCountList<Message>();
-            doBrowse(browsedMessages, getMaxExpirePageSize());
+    private final Runnable expireMessagesWork = () -> {
+        try {
```

Comment:
How about adding a fast-exit check here if there are zero messages and/or zero offline durable subscribers?

Reply:
See https://github.com/apache/activemq/pull/1423/commits/baff56ed8a4aaf0eb849f771c438284f79f1adfc

Reply:
Great!